internal/summon/command/temp_factory.go

```diff
@@ -63,24 +63,33 @@ func defaultTempPath(sharedMemoryDir string) string {
 }
 
 // Push creates a temp file with given value. Returns the path.
-func (tf *TempFactory) Push(value string) string {
-	f, _ := ioutil.TempFile(tf.path, ".summon")
-	defer f.Close()
+func (tf *TempFactory) Push(value string) (string, error) {
+	f, err := ioutil.TempFile(tf.path, ".summon")
+	defer func() {
+		_ = f.Close()
+	}()
+
+	if err != nil {
+		return "", err
+	}
+
+	_, err = f.Write([]byte(value))
+	if err != nil {
+		return "", err
+	}
```

Comment:
Couple of issues here:
- `f` must be a value before we set a `defer`
- `defer` can be inlined

```suggestion
func (tf *TempFactory) Push(value string) (string, error) {
    f, err := ioutil.TempFile(tf.path, ".summon")
    if err != nil {
        return "", err
    }
	
    defer f.Close()

    _, err = f.Write([]byte(value))
    if err != nil {
        return "", err
    }
```